hive/inference_image.py

```diff
@@ -0,0 +1,181 @@
+  if len(predict_input_filename_list) != len(predict_output_filename_list):
+    print("Unequal input and output file lists")
+    po.print_help()
+    sys.exit(1)
+
+  # build the model from a config file and a checkpoint file
+  model = init_segmentor(config_file, checkpoint_file, device='cuda:0')
+  # model = init_segmentor(config_file, checkpoint_file, device='cpu')
+
+  COLOR_DICT_m7 = {0: "Other_m7", 1: "Mobile_m7", 2: "Trees_m7", 3: "Ground_m7", 4: "Structure_m7", 5: "Building_m7", 6: "Water_m7"}
+  model.CLASSES = [val for key, val in COLOR_DICT_m7.items()]
+  print("Class map:", model.CLASSES)
+
+  # Should this always be 2?
+  upscale = 2.
+
+  for image_filename, out_filename in zip(predict_input_filename_list, predict_output_filename_list):
+    if fnmatch.fnmatch(image_filename, '*.jpg') or fnmatch.fnmatch(image_filename, '*.png'):
+      img_array = mmcv.imread(image_filename)
+      print("Running", image_filename, "with shape", img_array.shape)
+      image_meta = {}
+      image_meta['ori_shape'] = img_array.shape
+      img = mmcv.image.geometric.imrescale(img_array, upscale)
+      image_meta['pad_shape'] = img.shape
+      image_meta['img_shape'] = img.shape
+      image_meta['scale_factor'] = scale
+      image_meta['flip'] = False
+      image_meta['flip_direction'] = 'horizontal'
+      image_meta['filename'] = None
+      image_meta['ori_filename'] = None
+      image_meta['scale_factor'] = np.array([upscale, upscale, upscale, upscale], dtype=np.float32)
+      # TODO get these from the model config
+      image_meta['img_norm_cfg'] = {'mean': np.array([97.8711, 98.77822, 89.18051], dtype=np.float32), 'std': np.array([50.344364, 47.63134, 48.946396], dtype=np.float32), 'to_rgb': True}
+
+      # Prepare the image tensor
+      image = torch.from_numpy(img)
+      image = image[np.newaxis, :]
+      # Then to swap the axes as desired--note: _unsqueeze works fine here too
+      image = image.permute(0, 3, 1, 2)
+      image = torch.tensor(image, dtype=torch.float32)
+      # print("Input:", image.shape)
+
+      # Image tensor data type depends on cuda
+      device = next(model.parameters()).device
+      is_cuda = next(model.parameters()).is_cuda
+      print(device, is_cuda)
+      if is_cuda:
+        image = image.cuda()
+
+      # Run data but disable gradient
+      with torch.no_grad():
+        result = model.slide_inference(image, [image_meta], rescale=True)
+
+      # Get data from result
+      if is_cuda:
+        # pull tensor out of GPU into system ram and convert to numpy
+        output = np.array(result[0].permute(1, 2, 0).cpu())
+      else:
+        # Just convert to numpy
+        output = np.array(result[0].permute(1, 2, 0))
+
+      # Normalize each pixel val here
+      for iy in range(output.shape[0]):
+        for ix in range(output.shape[1]):
+          element = output[iy][ix]
+          min_val = np.min(element)
+          max_val = np.max(element)
+          range_vals = max_val - min_val
```

Comment:
Not sure if these are ints or floats, but should we instead
```suggestion
          range_vals = max_val - min_val + 1
```
?

If nothing else, it eliminates the potential divide-by-zero below.

Reply:
Great point on the divide by zero.  The data at this point are still floats.